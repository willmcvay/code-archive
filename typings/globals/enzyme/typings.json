{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4cc7cbabf6dce8e5e684622d538db6241fbc0a7b/enzyme/enzyme.d.ts",
    "raw": "registry:dt/enzyme#2.5.1+20161025153540",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4cc7cbabf6dce8e5e684622d538db6241fbc0a7b/enzyme/enzyme.d.ts"
  }
}
