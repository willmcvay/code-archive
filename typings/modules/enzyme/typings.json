{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/ce5eb0cc2c065553da248bab5f08f91575a37873/enzyme/index.d.ts",
    "raw": "registry:dt/enzyme#2.7.0+20170105200209",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/ce5eb0cc2c065553da248bab5f08f91575a37873/enzyme/index.d.ts"
  }
}
