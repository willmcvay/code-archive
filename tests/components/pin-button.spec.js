import React from 'react'
import { shallow } from 'enzyme'
import PinButton from '../../src/components/pin-button'

const defaultProps = {
  value: 1,
  numberTapped: jest.fn()
}

describe('PinButton', () => {
  it('should match a DOM snapshot', () => {
    expect(shallow(<PinButton {...defaultProps} />)).toMatchSnapshot()
  })

  it('should respond correctly to being clicked', () => {
    const component = shallow(<PinButton {...defaultProps} />)
    component
      .find('.number')
      .first()
      .simulate('click')

    expect(defaultProps.numberTapped).toHaveBeenCalledTimes(1)
  })
})
