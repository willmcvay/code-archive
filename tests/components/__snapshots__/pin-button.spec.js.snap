// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinButton should match a DOM snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PinButton
    numberTapped={[MockFunction]}
    value={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 1,
      "className": "number",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": 1,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 1,
        "className": "number",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": 1,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
